charts/spire/charts/spire-server/templates/configmap.yaml

```diff
@@ -32,59 +32,59 @@ server:
 
 plugins:
   DataStore:
-    - sql:
-        plugin_data:
-          {{ include "spire-server.datastore-config" . | nindent 10 }}
+    sql:
+      plugin_data:
+        {{ include "spire-server.datastore-config" . | nindent 8 }}
 
-  {{- with .Values.nodeAttestor.k8sPsat }}
+  {{- with .Values.nodeAttestor.k8s_psat }}
   {{- if eq (.enabled | toString) "true" }}
   NodeAttestor:
-    - k8s_psat:
-        plugin_data:
-          clusters:
-            {{ include "spire-lib.cluster-name" $root }}:
-              service_account_allow_list: {{ include "spire-server.serviceAccountAllowedList" $root | trim }}
+    k8s_psat:
+      plugin_data:
+        clusters:
+          {{ include "spire-lib.cluster-name" $root }}:
+            service_account_allow_list: {{ include "spire-server.serviceAccountAllowedList" $root | trim }}
   {{- end }}
   {{- end }}
 
   KeyManager:
-    - disk:
-        plugin_data:
-          keys_path: "/run/spire/data/keys.json"
+    disk:
+      plugin_data:
+        keys_path: "/run/spire/data/keys.json"
 
   Notifier:
-    - k8sbundle:
-        plugin_data:
-          namespace: {{ .Values.notifier.k8sbundle.namespace | default (include "spire-server.namespace" .) | quote }}
-          config_map: {{ include "spire-lib.bundle-configmap" . | quote }}
+    k8sbundle:
+      plugin_data:
+        namespace: {{ .Values.notifier.k8sbundle.namespace | default (include "spire-server.namespace" .) | quote }}
+        config_map: {{ include "spire-lib.bundle-configmap" . | quote }}
 
   {{- with .Values.upstreamAuthority.disk }}
   {{- if eq (.enabled | toString) "true" }}
   {{- $upstreamAuthorityUsed = add1 $upstreamAuthorityUsed }}
   UpstreamAuthority:
-    - disk:
-        plugin_data:
-          cert_file_path: "/run/spire/upstream_ca/tls.crt"
-          key_file_path: "/run/spire/upstream_ca/tls.key"
-          {{- if ne .secret.data.bundle "" }}
-          bundle_file_path: "/run/spire/upstream_ca/bundle.crt"
-          {{- end }}
+    disk:
```

Comment:
Isn't this breaking, I thought we required those `-` earlier.

Reply:
The macro makes it optional, so values merging works better.

Reply:
All the tests pass too